client/state/selectors/is-gutenberg-enabled.js

```diff
@@ -24,16 +22,35 @@ export const isGutenbergEnabled = ( state, siteId ) => {
 		return true;
 	}
 
-	if ( isEnabled( 'jetpack/gutenframe' ) ) {
-		if (
-			versionCompare(
-				get( getSiteOptions( state, siteId ), 'jetpack_version', 0 ),
-				'7.3-alpha',
-				'>='
-			)
-		) {
-			return isEnabled( 'gutenberg' ) && ! isVipSite( state, siteId );
+	// We do want Gutenframe flows for Jetpack and Atomic sites.
+	if ( isJetpackSite( state, siteId ) || isSiteAutomatedTransfer( state, siteId ) ) {
+		// But only once they have been updated to WordPress version 5.0 or greater since it will provide Gutenberg
+		// editor by default.
+		const wpVersion = getWordPressVersion( state, siteId );
+		if ( versionCompare( wpVersion, '5.0', '<' ) ) {
+			return false;
+		}
+
+		// But not if they activated the Classic Editor plugin (effectively opting out of Gutenberg).
+		if ( isPluginActive( state, siteId, 'classic-editor' ) ) {
+			return false;
 		}
+
+		// And also only if they have been updated to Jetpack 7.3 or greater since it will provide a way to handle the
+		// frame nonces verification.
+		if ( ! isJetpackMinimumVersion( state, siteId, '7.3-alpha' ) ) {
+			return false;
+		}
+
+		// But only if the site has a SSL cert since the browser cannot embed insecure content in a resource loaded
+		// over a secure HTTPS connection.
+		if ( ! isHttps( getSiteAdminUrl( state, siteId ) ) ) {
+			return false;
+		}
+
+		return (
+			isEnabled( 'jetpack/gutenframe' ) && isEnabled( 'gutenberg' ) && ! isVipSite( state, siteId )
```

Comment:
Are there Jetpack/Atomic sites that are also VIP sites?

Reply:
The `isJetpackSite()` check three lines down could probably be removed.

Reply:
> Are there Jetpack/Atomic sites that are also VIP sites?

No.

Reply:
I'm all for closing PRs as soon as possible, although here I was hoping we could remove the unnecessary checks first. Next time

Reply:
Removed on https://github.com/Automattic/wp-calypso/pull/32678